### ISSUE TITLE
Missing feature to define multiple waypoints and draw road

### ISSUE THREAD

**Issue by cbopp-art** [liodali/osm_flutter#104]
Would it possible to draw roads (add/remove waypoints) with tapping directly inside the map? Is there maybe at least a listener for tapping/click event on the map?
This would be a great feature because I need to draw routes by tapping/clicking on the map and finally drawing a road based on all the defined waypoints.

**Comment by liodali** [comment]
I already open discussion about `selectionPosition` #97  and I proposed to add longPressListen callback 
this feature i will added by the end of june when at least add drawRoad for ios support.
for drawRoad you can draw route with multple points just you need to add `interestPoints` attribute

**Comment by liodali** [comment]
I have another question
the method `drawRoadè  doesn't fill your need ? or you just want callback to get tapping from map ?

**Comment by cbopp-art** [comment]
You're right, I found a way using `mapController.selectPosition();` and `mapController.drawRoad()`. However, I had to fork the repo and change the API URL from https://router.project-osrm.org/route/v1/driving/ to https://routing.openstreetmap.de/ in order to get it working. Seems that the API source is broken or has changed.
Test this link and see that it is broken:
https://router.project-osrm.org/route/v1/driving/routed-car/route/v1/driving/8.2278234308,47.4159237209;8.2386224920,47.4168952180;8.2477985729,47.4235684965;8.2446774569,47.4285095564?alternatives=false&overview=full&steps=true

**Comment by liodali** [comment]
yeah, i will make new drawRoad more customizable 
and also i started creation the feature of tapping callback into the map(android)  but i'm facing small problem but it will be shipped by the next week

**Comment by liodali** [comment]
the osrm using /route/v1/diving 
but if you have your own routing api 
i will add new way as i told you that you will configure routing api in library

**Comment by liodali** [comment]
that was my fault when i upgraded osmbonuspack i didn't realize that thet start using https and they change endpoint
the new version 0.7.8+4 fix that problem #109

**Comment by liodali** [comment]
good news listener to SingleTap and LongPress is ready to be shipped in the next version
here example that you will find in readme
```dart
controller.listenerMapLongTapping.addListener(() {
      if(controller.listenerMapLongTapping.value!=null){
            //logic
      }
    });
    controller.listenerMapSingleTapping.addListener(() {
      if(controller.listenerMapSingleTapping.value!=null){
          //logic
      }
    });

```
and you can get last geoPoint tapped like this and don't forget to test if it null or not
```dart
final geoP = controller.listenerMapLongTapping.value;

```

**Comment by cbopp-art** [comment]
Hi @liodali 
This is amazing. Thank you! I will test it.

**Comment by cbopp-art** [comment]
Hi, It would be nice beeing able to define the route mode is for driving, walking etc. I can see that the API url has been completely removed now within this plugin. Any chance to define and customize the url?

**Comment by liodali** [comment]
* for osm routing i will add that just i need finish drawing road in ios side to implement that 
* for customizable url, that little complicated because i need to know if endpoint return the same as routing api of osm
and also I'm planning to support for geojson format that will make easly integrated in ios and android side from dart

**Comment by cbopp-art** [comment]
I cannot get it running. The listeners do work, but now drawRoad() seems to be broken.

**Comment by liodali** [comment]
can you provide me more information

**Comment by cbopp-art** [comment]
I figured out now what does not work. The listeners only work once, after changing screen and returning again, the listeners do not work anymore. Would you be able to reproduce this behavior?

**Comment by liodali** [comment]
did you dispose the screen when you go the next screen or you keep it alive
but I idea about the problem  I will fix it

**Comment by liodali** [comment]
when you change screen id of method channel change and in dart side i already manage to listener to multiple id of method channel but it seems that it dosn't work properly i will fix it

**Comment by liodali** [comment]
I fix the problem i will publish new version 
the problem that when you change the screen and go back to the map i forget to re init stream because i already disposed

**Comment by liodali** [comment]
check the new version `0.7.9+2`

**Comment by cbopp-art** [comment]
Great, thank you. It works now. I would like to add a first marker for drawing a road using the coordinates after tapping on the map using this listener, similar to what "await controller.selectPosition()" basically would do. It works, but i cannot define the marker icon for that. However, I think this is another topic :)

**Comment by liodali** [comment]
yeah  i will add new method addMarker for that purpose

**Comment by liodali** [comment]
i added method `addMarker` to be used with Taplistener or in any other use case
it will be available in version 0.12.0
